packages/astro/src/assets/internal.ts

```diff
@@ -104,7 +106,15 @@ export async function generateImage(
 		clientRoot = buildOpts.settings.config.outDir;
 	}
 
-	const fileData = await fs.promises.readFile(new URL('.' + options.src.src, serverRoot));
+	const fileData = await fs.promises.readFile(
+		new URL(
+			'.' +
+				prependForwardSlash(
+					join(buildOpts.settings.config.build.assets, basename(options.src.src))
```

Comment:
Could you alias `options.src.src` to some variable before using it? Reading it here I don't know what it points to. Give it some descriptive name.

Also, this needs to use `path.posix.join` or there will be backslashes on Windows.